### PR TITLE
Add tests associated to Task

### PR DESCRIPTION
**Pull request by johnrhimawan**
Currently, ```InTrack``` does not provide testing to ```Task``` and its associated features. Let's add tests to further improve our code coverage.

### CODE REVIEW

File: src/test/java/seedu/intrack/model/internship/TaskTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.intrack.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class TaskTest {
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Task(null, null));
+    }
```
Comment (chantellyu) [inline src/test/java/seedu/intrack/model/internship/TaskTest.java:12]: Would it be better to test the two null inputs separately? Following the "no more than one invalid input in a test case" heuristic.

File: src/test/java/seedu/intrack/model/internship/TaskTest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    @Test
+    public void constructor_invalidTask_throwsIllegalArgumentException() {
+        String invalidTaskName = "";
+        String invalidTaskTime = "TODAY";
+        assertThrows(IllegalArgumentException.class, () -> new Task(invalidTaskName, invalidTaskTime));
+    }
```
Comment (chantellyu) [inline src/test/java/seedu/intrack/model/internship/TaskTest.java:19]: Still on the idea same idea as above, would it be better to separate invalid inputs for taskName and taskTime?

File: src/test/java/seedu/intrack/model/internship/TaskTest.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    public void isValidTask() {
+        // null task
+        assertThrows(NullPointerException.class, () -> Task.isValidTask(null));
+
+        // valid tasks
+        assertTrue(Task.isValidTask(new Task("Online Assessment", "10-10-2023 10:00")));
+        assertTrue(Task.isValidTask(new Task("-", "05-11-2022 11:00"))); // one character
+        assertTrue(Task.isValidTask(new Task("The quick brown fox jumped over the lazy dog while testing"
+                + "the code in his project that is very very long", "20-10-2022 15:00"))); // long taskName
+    }
```
Comment (chantellyu) [inline src/test/java/seedu/intrack/model/internship/TaskTest.java:31]: Maybe you can consider splitting these into individual test cases for better coherency and ease in reading the test case file?

**Review by chantellyu — COMMENTED** [review]
Good idea to abstract out the default taskName as a constant! Just some suggestions you can consider for the structure of the test case file.

File: src/test/java/seedu/intrack/model/internship/TaskTest.java (hunk 1 above)
Comment (johnrhimawan) [reply src/test/java/seedu/intrack/model/internship/TaskTest.java:12]: Thank you for the suggestion, I will update the tests accordingly.

File: src/test/java/seedu/intrack/model/internship/TaskTest.java (hunk 2 above)
Comment (johnrhimawan) [reply src/test/java/seedu/intrack/model/internship/TaskTest.java:19]: Thank you for pointing that out, I will update the test case now.

**Pull request merged**